launcher/container_runner.go

```diff
@@ -194,43 +211,114 @@ func (r *ContainerRunner) measureContainerClaims(ctx context.Context) error {
 		}
 	}
 
-	contianerSpec, err := r.container.Spec(ctx)
+	containerSpec, err := r.container.Spec(ctx)
 	if err != nil {
 		return err
 	}
-	for _, arg := range contianerSpec.Process.Args {
+	for _, arg := range containerSpec.Process.Args {
 		if err := r.attestAgent.MeasureEvent(cel.CosTlv{EventType: cel.ArgType, EventContent: []byte(arg)}); err != nil {
 			return err
 		}
 	}
-	for _, env := range contianerSpec.Process.Env {
+	for _, env := range containerSpec.Process.Env {
 		if err := r.attestAgent.MeasureEvent(cel.CosTlv{EventType: cel.EnvVarType, EventContent: []byte(env)}); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// Run the container
-// Doesn't support container restart yet
-// Container output will always be redirected to stdio for now
-func (r *ContainerRunner) Run(ctx context.Context) error {
-	if err := r.measureContainerClaims(ctx); err != nil {
-		return fmt.Errorf("failed to measure container claims: %v", err)
+func getTTL(token []byte) (time.Duration, error) {
```

Comment:
I think it might make more sense to just have `getTTL` be `getExpiration` and return the `time.Time` that the token expires. Then all the expiration calculations can be done at the end of `refreshToken`. Something like:

```go
func getExpiration(token []byte) (time.Time, error) {
	claims := &jwt.RegisteredClaims{}
	_, _, err := new(jwt.Parser).ParseUnverified(string(token), claims)
	if err != nil {
		return time.Time{}, fmt.Errorf("failed to parse token: %w", err)
	}
	return claims.ExpiresAt.Time, nil
}

func (r *ContainerRunner) refreshToken(ctx context.Context) (time.Duration, error) {
	...
	exp, err := getExpiration(token)
	if err != nil {
		return 0, fmt.Errorf("failed to get token expiration: %v", err)
	}
	return time.Duration(float64(time.Until(exp)) * defaultRefreshMultiplier), nil
}
```

Reply:
I ended removing the `getExpiration` method since most of the functionality was moved out at that point, and it's only used once in `refreshToken` so it didn't seem necessary to have a separate method.